lib/zlib.js

```diff
@@ -78,6 +79,13 @@ function isInvalidStrategy(strategy) {
 }
 
 function zlibBuffer(engine, buffer, callback) {
+  // Streams do not support non-Buffer Uint8Arrays yet. Convert it to a
+  // Buffer without copying.
+  if (isUint8Array(buffer) &&
+      !(Object.getPrototypeOf(buffer) === Buffer.prototype ||
+        buffer instanceof Buffer))
+    buffer = Buffer.from(buffer.buffer, buffer.byteOffset, buffer.byteLength);
```

Comment:
@bnoordhuis I believe it’s because `instanceof` can be quite slow, so in the exceedingly common case the `getPrototypeOf` check suffices. But you’re right, I’d just drop the `instanceof` part.